cpp/include/cudf/table/row_operators.cuh

```diff
@@ -318,9 +360,39 @@ class element_relational_comparator {
                               rhs.element<Element>(rhs_element_index));
   }
 
+  /**
+   * @brief Performs a relational comparison between the specified elements
+   *
+   * @param lhs_element_index The index of the first element
+   * @param rhs_element_index The index of the second element
+   * @param nan_result specifies what value should be returned if either element is `nan`
+   * @return Indicates the relationship between the elements in
+   * the `lhs` and `rhs` columns.
+   */
+  template <typename Element,
+            std::enable_if_t<cudf::is_relationally_comparable<Element, Element>()>* = nullptr>
+  __device__ weak_ordering operator()(size_type lhs_element_index,
+                                      size_type rhs_element_index,
+                                      weak_ordering nan_result) const noexcept
```

Comment:
Can't this parameter instead be made state inside the `element_relational_comparator`?